### DISCUSSION TITLE
Reading DB credentials from env

### DISCUSSION THREAD

**Discussion by Cavdy** [prisma/prisma#8546] [Q&A] [answered]
Hey... I have a project I am building, in the installation page the user gets an input field to input their database credentials and I save it in a .env file. When the user puts in the wrong database credential on the first try, the Prisma schema gets it and throws an error which is great but when the user puts in the correct credential on the second try Prisma still use the old credentials even though the .env file now has the right credentials. How do I make Prisma read the .env file again to get the new credentials?

Right now I am directly saving the DB credentials to the schema file temporary

**Comment by ryands17** [comment, marked as answer]
@Cavdy 👋 

If you change the env variable, you need to restart your server as once variables from `.env` are read their value stays the same throughout the application.

**Reply by Cavdy** [reply]
Yeah that's true... Thanks @ryands17 

Currently I'm saving the credentials directly to the schema file because I don't want the user to experience downtime while restarting the server... Is this safe and good?

**Reply by ryands17** [reply]
You can also directly provide the credentials [here](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#datasources) under `datasource` and `PrismaClient` will use that directly.

**Reply by Cavdy** [reply]
Thanks @ryands17

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
